test/generators/typescript/preset/MarshallingPreset.spec.ts

```diff
@@ -0,0 +1,207 @@
+    
+      get sTestPatternProperties(): Map<String, string> | undefined { return this._sTestPatternProperties; }
+      set sTestPatternProperties(sTestPatternProperties: Map<String, string> | undefined) { this._sTestPatternProperties = sTestPatternProperties; }
+    
+      public marshal() : string {
+        let json = '{'
+        if(this.stringProp !== undefined) {
+          json += `"stringProp": ${JSON.stringify(this.stringProp)},`; 
+        }
+        if(this.numberProp !== undefined) {
+          json += `"numberProp": ${JSON.stringify(this.numberProp)},`; 
+        }
+        if(this.objectProp !== undefined) {
+          json += `"objectProp": ${this.objectProp.marshal()},`; 
+        }
+    
+        if(this.sTestPatternProperties !== undefined) { 
+          for (const [key, value] of this.sTestPatternProperties.entries()) {
+            //Only render pattern properties which are not already a property
+            if(Object.keys(this).includes(String(key))) continue;
+            json += `"${key}": ${JSON.stringify(value)},`;    
+          }
+        }
+    
+        if(this.additionalProperties !== undefined) { 
+          for (const [key, value] of this.additionalProperties.entries()) {
+            //Only render additionalProperties which are not already a property
+            if(Object.keys(this).includes(String(key))) continue;
+            json += `"${key}": ${JSON.stringify(value)},`;    
+          }
+        }
+        //Remove potential last comma 
+        return `${json.charAt(json.length-1) === ',' ? json.slice(0, json.length-1) : json}}`;
+      }
+    
+      public static unmarshal(json: string | object): Test {
+        const obj = typeof json === "object" ? json : JSON.parse(json);
+        const instance = new Test({});
+    
+        if (obj.stringProp !== undefined) {
+          instance.stringProp = obj.stringProp;
+        }
+        if (obj.numberProp !== undefined) {
+          instance.numberProp = obj.numberProp;
+        }
+        if (obj.objectProp !== undefined) {
+          instance.objectProp = NestedTest.unmarshal(obj.objectProp);
+        }
+    
+        //Not part of core properties
+        for (const [key, value] of Object.entries(obj).filter((([key,]) => {return !["stringProp","numberProp","objectProp"].includes(key);}))) {
+          //Check all pattern properties
+          if (key.match(new RegExp('^S(.?)test'))) {
+            if (instance.sTestPatternProperties === undefined) {instance.sTestPatternProperties = new Map();}
+            instance.sTestPatternProperties.set(key, value as any);
+            continue;
+          }
+    
+          if (instance.additionalProperties === undefined) {instance.additionalProperties = new Map();}
+          instance.additionalProperties.set(key, value as any);
+        }
```

Comment:
You should split this for to the two for, one for additionalProperties and one for patternProperties

Reply:
I don't think I can do that as I default to adding the property to `additionalProperties` property if not part of pattern properties.

Reply:
Ahh yes, you're right. It can stay.